From bc6d2af83a53cf7bd0db285ba2a5714448207f87 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <gqjiang@suse.com>
Date: Fri, 7 Jul 2017 10:42:29 +0800
Subject: [PATCH] dm: fix second blk_delay_queue() parameter to be in msec
 units not
References: bsc#1047670
Patch-Mainline: v4.8-rc1
Git-commit: bd9f55ea1cf6e14eb054b06ea877d2d1fa339514

This commit is based on below upstream commit, and backport it from
12sp3 to 12sp2, for sp2, the difference is blk_delay_queue is called
in dm.c since dm-rq.c is not created at that time.

Commit d548b34b062 ("dm: reduce the queue delay used in dm_request_fn
from 100ms to 10ms") always intended the value to be 10 msecs -- it
just expressed it in jiffies because earlier commit 7eaceaccab ("block:
remove per-queue plugging") did.

Signed-off-by: Tahsin Erdogan <tahsin@google.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Fixes: d548b34b062 ("dm: reduce the queue delay used in dm_request_fn from 100ms to 10ms")
Cc: stable@vger.kernel.org # 4.1+ -- stable@ backports must be applied to drivers/md/dm.c
Acked-by: Hannes Reinecke <hare@suse.de>
[gqjiang: backport from 12sp3 to 12sp2]
Signed-off-by: Guoqing Jiang <gqjiang@suse.com>
---
 drivers/md/dm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index 1c29149..f1b7e9c 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -2256,7 +2256,7 @@ static void dm_request_fn(struct request_queue *q)
 		     md_in_flight(md) && rq->bio && rq->bio->bi_vcnt == 1 &&
 		     md->last_rq_pos == pos && md->last_rq_rw == rq_data_dir(rq)) ||
 		    (ti->type->busy && ti->type->busy(ti))) {
-			blk_delay_queue(q, HZ / 100);
+			blk_delay_queue(q, 10);
 			return;
 		}
 
-- 
2.10.0

